### PR TITLE
docs(encryption): threat model, format/usage docs, final edge-case tests (M5 of #117)

### LINKED ISSUE #117: feat: optional archive encryption-at-rest (--encrypt)
## Summary
Add optional encryption for `.kshrk` archives so sensitive cluster captures can be stored and shared safely. Captures can contain Secrets, tokens, and PII; redaction (#redact) reduces but does not eliminate sensitivity.

## Why
The core workflow is handing a capture to someone else. Encryption-at-rest makes that safe by default-able, and complements the existing redaction feature.

## Scope (for discussion)
- `kshrk capture --encrypt` / `kshrk redact --encrypt` to produce an encrypted archive; `open`/`ui`/`inspect`/`diff` transparently decrypt when given a key/passphrase.
- Key options: passphrase (scrypt/argon2 + AEAD) and/or recipient public keys (age-style). Prefer a well-reviewed library over hand-rolled crypto.
- Clear UX when a key is required/missing; never write plaintext temp files.
- Interplay with the ZIP+Zstd format: encrypt the whole archive (simplest) vs per-entry (streamable) — decide in design.

## Acceptance criteria
- Round-trip: encrypt on write, decrypt on every read path, with a passphrase.
- No plaintext spillage (temp files, logs).
- Documented threat model and key handling.

## Notes
- Heaviest of the v1.0 polish items (net-new crypto, security-sensitive) — easy to defer to a later milestone if v1.0 scope tightens. Flagged here so it's tracked.

## Milestone
v1.0 (candidate — deferrable).

### PR DESCRIPTION
**Pull request by phenixblue**
## Summary

Fifth milestone of optional archive encryption-at-rest ([#117](https://github.com/phenixblue/k8shark/issues/117)): **documentation and a final edge-case test pass.** M1–M4 already shipped the full feature (core crypto, `capture --encrypt`, reader-side `--decrypt-*`, and public-key recipients); this closes the gap where encryption existed but wasn't documented anywhere. **Update:** review of this PR surfaced one genuine, user-visible bug fix along the way — see below.

## Docs

- **New `docs/encryption-threat-model.md`** — what's protected (contents, structure, tamper detection via AEAD) vs. explicitly out of scope (file existence/approximate size, a compromised host while the archive is open, recipient revocation, old `kshrk` binaries getting a raw error), the key-handling rules the CLI enforces (passphrase material never through Viper/`config.yaml`, treat identity files like SSH keys, passphrase/recipient mutual exclusion), a passphrase-vs-recipients comparison table, and today's re-key workaround via `kshrk redact` (with its `metadata.redacted=true` mislabeling caveat, pending the planned M6 `encrypt`/`decrypt` commands).
- **`docs/archive-format.md`** — new "Encryption" section: the whole-archive age envelope design, the header-sniff detection mechanism (not a metadata field, since `metadata.json` is itself ciphertext), why `format_version` is untouched, and what the envelope hides vs. its residual leak (ciphertext size, that it's an age file at all). Cross-linked with the threat-model doc.
- **`docs/usage.md`** — new "Encryption" section with capture examples for both key modes, the shared `--decrypt-*` flags that work identically across every read command, and combining encryption with redaction — plus `--encrypt-*`/`--decrypt-*` rows added to the existing Capture/Redact flags tables.
- **README** — doc-index table gains the threat-model doc.

## Bug fix found during review: `kshrk replay` never decrypted

Writing "every read command decrypts transparently" in the docs prompted a check against the actual code, which turned up a real gap: `kshrk replay` (distinct from `kshrk ui`'s `--ui` replay mode, which *was* wired in M3) never called the identity resolver, so `server.Replay` always ran with `nil` identities. Any encrypted archive failed with a generic "supply a decryption key" error regardless of what `--decrypt-passphrase-file`/`--decrypt-identity-file`/`$KSHRK_DECRYPT_PASSPHRASE` contained — the flags were silently ignored.

Fixed in `cmd/replay.go` (now resolves and passes identities, matching every other read command), confirmed no other command has the same gap by sweeping all `archive.Open`/`server.Open`/`server.Replay` call sites in `cmd/`, and added a regression test that specifically distinguishes "wrong passphrase" from "no passphrase" — the signal that proves the flag is actually being read rather than discarded.

## Tests

The feature was already heavily tested through M1–M4 (round-trips, tamper detection, wrong-key, non-TTY, concurrency, passphrase↔recipient re-keying, etc.). This fills the two real gaps rather than padding coverage:

- **`internal/server/encrypt_test.go`** — full HTTP round trips through an encrypted archive via `server.Open` and `server.Replay`. This is the long-lived-holder path behind `open`/`ui`; prior coverage only unit-tested the CLI resolver and single-shot commands like `inspect`, so the actual "decrypt once at launch, serve many requests" flow was untested. Includes a clean-failure case with no identities.
- **`cmd/diff_test.go`** — an end-to-end two-archive diff with only `--before` encrypted, exercising the M3 multi-path-prompt fix through the *full* diff pipeline (not just the resolver in isolation, which is all the original fix's test covered).

`make fmt`, `make lint-ci`, full `make test`, and `go mod tidy` (idempotent) all clean.

## Note for reviewers

Does **not** close #117 — M5 of 5 in the original plan. A follow-up **M6** (standalone `kshrk encrypt`/`kshrk decrypt` commands, agreed with the maintainer after M4) remains, referenced in the threat-model doc's "Rotating keys today" section as the reason today's re-key path is a `redact`-based workaround.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/server/encrypt_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	if err != nil {
+		t.Fatalf("server.Open: %v", err)
+	}
+	defer srv.Shutdown()
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, // #nosec G402 — test only
+		},
+	}
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply internal/server/encrypt_test.go]: Fixed in efda510 — both tests now use a pinnedClient helper that verifies the server's self-signed cert via a RootCAs pool built from srv.CertPEM(), instead of InsecureSkipVerify.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/server/encrypt_test.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+// (not a raw zip-parse error) when the archive is encrypted and no key is
+// supplied.
+func TestServer_Open_EncryptedArchiveNoIdentities(t *testing.T) {
+	archivePath := buildEncryptedTestArchive(t)
+
+	_, err := Open(OpenOptions{ArchivePath: archivePath})
+	if err == nil {
+		t.Fatal("server.Open on encrypted archive with no identities succeeded, want error")
+	}
+}
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply internal/server/encrypt_test.go:151]: Fixed in efda510 — the test now asserts the error contains "is encrypted", matching the comment's intent and catching a regression to a raw zip-parse error.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: cmd/diff_test.go [hunk 3]
```diff
@@ -77,9 +79,74 @@ func newTestDiffCommand() *cobra.Command {
...
+	passFile := filepath.Join(t.TempDir(), "pass.txt")
+	if err := os.WriteFile(passFile, []byte(passphrase), 0o600); err != nil {
+		t.Fatalf("WriteFile: %v", err)
+	}
+
+	cmd := newTestDiffCommand()
+	_ = cmd.Flags().Set("before", before)
+	_ = cmd.Flags().Set("after", after)
+	_ = cmd.Flags().Set("output", "json")
+	_ = cmd.Flags().Set("decrypt-passphrase-file", passFile)
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply cmd/diff_test.go]: Fixed in efda510 — all four flag Set calls now check the returned error and t.Fatalf immediately, matching this file's existing pattern.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/server/encrypt_test.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+	}
+	items, ok := pods["items"].([]any)
+	if !ok || len(items) == 0 {
+		t.Fatal("expected at least one pod in items")
+	}
+	item := items[0].(map[string]any)
+	meta := item["metadata"].(map[string]any)
+	if meta["name"] != "nginx" {
+		t.Errorf("expected pod name=nginx, got %v", meta["name"])
+	}
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply internal/server/encrypt_test.go]: Fixed in efda510 — the items[0] and metadata type assertions now check their ok bool and t.Fatalf with the actual value on failure instead of risking a panic.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: docs/archive-format.md [hunk 5]
```diff
@@ -2,7 +2,10 @@
 
 A k8shark capture is a `.kshrk` file: a ZIP container whose entries are
 individually Zstandard-compressed JSON (except `metadata.json`, which is stored
-uncompressed for fast header reads). It can be listed with any ZIP tool.
+uncompressed for fast header reads). It can be listed with any ZIP tool —
+unless it was written with `--encrypt`/`--encrypt-recipient`, in which case the
+whole file is a single [age](https://age-encryption.org/v1) envelope around
+this same ZIP layout; see [Encryption](#encryption) below.
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply docs/archive-format.md]: Fixed in 667e971 — both the intro (line 8) and the Encryption section (line 232, also flagged as suppressed low-confidence) now list all four flags (--encrypt, --encrypt-passphrase-file, --encrypt-recipient, --encrypt-recipients-file) as enabling encryption, not just the two mode-selecting flags.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: docs/encryption-threat-model.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
+# Encryption threat model
+
+`.kshrk` captures can contain Kubernetes Secret values, tokens, and other
+cluster PII. `kshrk capture --encrypt` / `--encrypt-recipient` (and the
+equivalent `kshrk redact` flags) let you write an archive as a single
+[age](https://age-encryption.org/v1) envelope. This page states plainly what
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply docs/encryption-threat-model.md]: Fixed in 667e971 — the opening paragraph now lists all four flags that enable encryption, not just --encrypt / --encrypt-recipient.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: docs/usage.md [hunk 7]
```diff
@@ -727,13 +733,100 @@ Examples:
...
 
 ---
 
+## Encryption
+
+Captures can contain Secret values and other sensitive cluster data.
+`kshrk capture` (and `kshrk redact`) can write the archive as a single
+encrypted [age](https://age-encryption.org/v1) envelope, and every command
+that reads a `.kshrk` archive — `inspect`, `open`, `ui`, `diff`, `query`,
+`transitions`, `diagnose`, `redact` — decrypts it transparently given a key.
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply docs/usage.md]: Good catch — this was a real functional gap, not just a docs issue. Fixed in 844e498: cmd/replay.go now resolves --decrypt-* identities and passes them into server.Replay (it was the only read command missing this wiring; swept all cmd/*.go archive/server-open call sites to confirm). Added a regression test that specifically distinguishes "wrong passphrase" from "no passphrase" to prove the flag is actually used, not silently ignored.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/server/encrypt_test.go [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+		t.Fatalf("NewScryptRecipient: %v", err)
+	}
+	r.SetWorkFactor(10)
+	sw, err := archive.NewEncryptedStreamWriter(outPath, []age.Recipient{r})
+	if err != nil {
+		t.Fatalf("NewEncryptedStreamWriter: %v", err)
+	}
+	if err := sw.WriteRecord(rec); err != nil {
+		t.Fatalf("WriteRecord: %v", err)
+	}
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply internal/server/encrypt_test.go:62]: Fixed in 840647e — buildEncryptedTestArchive now defers sw.Abort() right after creation, matching the pattern in internal/redact/encrypt_test.go.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: cmd/diff_test.go [hunk 9]
```diff
@@ -77,9 +79,82 @@ func newTestDiffCommand() *cobra.Command {
...
+		t.Fatalf("NewScryptRecipient: %v", err)
+	}
+	r.SetWorkFactor(10)
+	sw, err := archivepkg.NewEncryptedStreamWriter(out, []age.Recipient{r})
+	if err != nil {
+		t.Fatalf("NewEncryptedStreamWriter: %v", err)
+	}
+	if err := sw.WriteRecord(rec); err != nil {
+		t.Fatalf("WriteRecord: %v", err)
+	}
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply cmd/diff_test.go:116]: Fixed in 840647e — buildEncryptedDiffArchive now defers sw.Abort() right after creation, matching the pattern in internal/redact/encrypt_test.go.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: docs/archive-format.md [hunk 10]
```diff
@@ -2,7 +2,12 @@
 
 A k8shark capture is a `.kshrk` file: a ZIP container whose entries are
 individually Zstandard-compressed JSON (except `metadata.json`, which is stored
-uncompressed for fast header reads). It can be listed with any ZIP tool.
+uncompressed for fast header reads). It can be listed with any ZIP tool —
+unless it was written with any of `capture`'s encryption flags
+(`--encrypt`, `--encrypt-passphrase-file`, `--encrypt-recipient`, or
+`--encrypt-recipients-file`), in which case the whole file is a single
+[age](https://age-encryption.org/v1) envelope around this same ZIP layout; see
+[Encryption](#encryption) below.
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply docs/archive-format.md:10]: Fixed in 840647e — the intro no longer attributes the --encrypt* flags to `capture` specifically; it now says they're available on both `kshrk capture` and `kshrk redact`, since redact writes the same encrypted format.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: docs/usage.md [hunk 11]
```diff
@@ -727,13 +733,101 @@ Examples:
...
+kshrk diff --before a.kshrk --after b.kshrk --decrypt-passphrase-file ./pass.txt
+```
+
+A wrong passphrase or key produces a clean error (`incorrect passphrase or
+key`), not a raw crypto failure.
+
+### Combining with redaction
+
+`kshrk redact` reads an encrypted source with the flags above and can
+re-encrypt the output with `--encrypt*`:
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply docs/usage.md]: Fixed in 840647e — both occurrences (this one and line 825, also flagged as suppressed low-confidence) now use `--encrypt-*`, consistent with `--decrypt-*` and unambiguous against the literal --encrypt flag.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: cmd/replay.go [hunk 12]
```diff
@@ -79,6 +79,11 @@ func runReplay(cmd *cobra.Command, args []string) error {
 		writable = true
 	}
 
+	identities, err := resolveDecryptIdentities(cmd, args[0])
+	if err != nil {
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply cmd/replay.go:85]: Fair point — updated the PR description to call out the replay fix explicitly as a user-visible bug fix (new "Bug fix found during review" section) rather than leaving the "no behavior changes" framing standing.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
